onward/app/weather/geo/CitiesLookUpV2.scala

```diff
@@ -0,0 +1,85 @@
+package weather.geo
+
+import common.ResourcesHelper
+
+import scala.io.{BufferedSource, Source}
+import scala.util.Try
+
+case class CityRef(city: String, region: String, country: String)
+
+object CityRef {
+  def makeFixedCase(city: String, region: String, country: String): CityRef =
+    CityRef(city.toUpperCase(), region.toUpperCase(), country.toUpperCase())
+}
+
+object CitiesCsvLine {
+  implicit class RichString(s: String) {
+    def withoutQuotes: String = s.stripPrefix("\"").stripSuffix("\"")
+  }
+
+  def unapply(line: String): Option[CitiesCsvLine] = {
+    line.split(",", -1) match {
+      case Array(
+            geonamesId,
+            city,
+            region,
+            country,
+            latitude,
+            longitude,
+          ) =>
+        Try {
+          CitiesCsvLine(
+            geonamesId.toInt,
+            city.withoutQuotes,
+            region.withoutQuotes,
+            country.withoutQuotes,
+            latitude.toDouble,
+            longitude.toDouble,
+          )
+        }.toOption
+
+      case _ => None
+    }
+  }
+}
+
+case class CitiesCsvLine(
+    geonamesId: Int,
+    city: String,
+    region: String,
+    country: String,
+    latitude: Double,
+    longitude: Double,
```

Comment:
I think a `Float` would be plenty, here.

Reply:
Just thought it was worth knowing that 6 decimal places gives you 1m precision.

https://gis.stackexchange.com/a/2964/95613